src/internal/components/cartesian-chart/left-labels.tsx

```diff
@@ -42,15 +45,17 @@ function LeftLabels({
 
   const yOffset = axis === 'x' && scale.isCategorical() ? Math.max(0, scale.d3Scale.bandwidth() - 1) / 2 : 0;
 
-  const cacheRef = useRef<{ [label: string]: number }>({});
+  const labelToHeightCache = useRef<{ [label: string]: number }>({});
+  const labelToWidthCache = useRef<{ [label: string]: number }>({});
   const getLabelSpace = (label: string) => {
-    if (cacheRef.current[label] !== undefined) {
-      return cacheRef.current[label];
+    if (labelToHeightCache.current[label] !== undefined) {
+      return labelToHeightCache.current[label];
     }
     if (virtualTextRef.current && virtualTextRef.current.getBBox) {
       virtualTextRef.current.textContent = label;
-      cacheRef.current[label] = virtualTextRef.current.getBBox().height;
-      return cacheRef.current[label];
+      labelToHeightCache.current[label] = virtualTextRef.current.getBBox().height;
+      labelToWidthCache.current[label] = virtualTextRef.current.getBBox().width;
```

Comment:
No need to call `virtualTextRef.current.getBBox()` twice. Could `labelToHeightCache` and `labelToWidthCache` be merged in the same object? Maybe we can just cache the raw boundingBox?